README.md

```diff
@@ -9,9 +9,84 @@
 
 Shared [Travis CI](https://travis-ci.com/), [GitHub Actions](https://docs.github.com/en/actions) and [pre-commit](https://pre-commit.com/) configuration files plus misc tools.
 
-## Travis
+## GitHub Actions
 
-## Migrate from Travis to GitHub Action
+### Java setup
+
+#### Setup JDK
+
+[actions/setup-java](https://github.com/actions/setup-java) should be used, here is a sample usage:
+
+```yml
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'temurin'
+          cache: 'maven'
+```
+
+#### Setup Maven Credentials
+
+Credentials should be already available via organization secrets, otherwise they would need to be
+provided as repository secrets.
+
+Since repositories hold a `settings.xml` file at the root with environment variables `MAVEN_USERNAME` and
+`MAVEN_USERNAME` filled for the username and password, only a mapping of variables is needed:
+
+```yml
+      - name: Build with Maven
+        run: mvn --settings settings.xml [...]
+        env:
+          MAVEN_USERNAME: ${{ secrets.NEXUS_USERNAME }}
+          MAVEN_USERNAME: ${{ secrets.NEXUS_PASSWORD }}
+```
+
+Alternatively, the [s4u/maven-settings-action](https://github.com/s4u/maven-settings-action) could be used.
+
+#### Setup Maven Build Options
+
+Maven build options can be shared for a given step on the mvn command line, or extracted as environment variables.
+
+Sample usage:
+
+```yml
+      - name: Test with Maven
+        run: mvn verify ${{ env.MAVEN_CLI_OPTS }}
+        env:
+          MAVEN_CLI_OPTS: --show-version -Ddocker.skip -Dlogging.root.level=off -Dspring.main.banner-mode=off
+```
+
+When deploying in a second step, these variables can be shared:
+
+```yml
+env:
+  MAVEN_CLI_OPTS: --show-version -Dlogging.root.level=off -Dspring.main.banner-mode=off
+
+[...]
+
+      - name: Test with Maven
+        run: mvn verify ${{ env.MAVEN_CLI_OPTS }}
+      - name: Deploy with Maven
+        run: mvn deploy ${{ env.MAVEN_CLI_OPTS }} -DskipTests
+```
+
+When migrating from Travis, depending on the previous configuration, docker.skip and docker.tag properties might need
```

Comment:
I would remove this as it's relevant for the AAE build only and probably better refactored to use standard docker actions